docs/content/concepts/sui-move-concepts/conventions.mdx

```diff
@@ -1,494 +1,458 @@
 
-Name the functions that create data structures as `empty`.
+Try to sort your functions by their purpose and according to the user flow to improve readability, you can also use explicit function names like `admin_set_fees` to make it clear what the function does.
 
-```move
-module conventions::collection {
+Ideally, test functions should only consists of `[test_only]` helpers for the actual tests that should be located in the tests directory.
 
-    struct Collection has copy, drop, store {
-        bits: vector<u8>
-    }
+Imports can be grouped by dependency, for example:
 
-    public fun empty(): Collection {
-        Collection {
-            bits: vector[]
-        }
-    }
-}
+```move
+use std::string::String;
+use sui::{
+    coin::Coin,
+    balance,
+    table::Table
+};
+use my_dep::battle::{Battle, Score};
 ```
 
-## New function
+## Naming Conventions
+
+### Constants
 
-Name the functions that create objects as `new`.
+Constants should be in uppercase and use snake case. Errors are specific constants that use PascalCase and start with an E. Make them descriptive.
 
 ```move
-module conventions::object {
+module conventions::constants;
 
-    use sui::object::{Self, UID};
-    use sui::tx_context::TxContext;
+const MAX_NAME_LENGTH: u64 = 64;
+// this is right
+const EInvalidName: u64 = 0;
+// this is wrong
+const E_INVALID_NAME: u64 = 0;
+```
 
-    struct Object has key, store {
-        id: UID
-    }
+### Structs
 
-    public fun new(ctx:&mut TxContext): Object {
-        Object {
-            id: object::new(ctx)
-        }
-    }
-}
-```
+To anticipate for future features, struct must be declared with the `public` modifier. Abilities must always be declared in this order: `key`, `copy`, `drop`, `store`. Do not use 'potato' in the name of structs. The lack of abilities define it as a potato pattern.
 
-## Shared objects
+Structs support positional fields that can be used for simple wrappers, dynamic fields keys or as tuples.
 
-Library modules that share objects should provide two functions: one to create the object and another to share it. It allows the caller to access its UID and run custom functionality before sharing it.
+Some the structs will be used to emit events, these structs should be named with the `Event` suffix.
 
 ```move
-module conventions::profile {
+module conventions::request;
 
-    use sui::object::{Self, UID};
-    use sui::tx_context::TxContext;
-    use sui::transfer::share_object;
+// dynamic field keys
+public struct ReceiptKey(ID) has copy, drop, store;
 
-    struct Profile has key {
-        id: UID
-    }
+// dynamic field
+public struct Receipt<Data> has key, store {
+    id: UID,
+    data: Data
+}
 
-    public fun new(ctx:&mut TxContext): Profile {
-        Profile {
-            id: object::new(ctx)
-        }
-    }
+// right naming
+public struct Request {}
```

Comment:
I think we'd have a weak preference for `public struct Request()` for this now that this is valid syntax.